### PR TITLE
Added framework for loading all Conditions and Effects

### PR DESCRIPTION
**Pull request by Hop311**
- Added base `Script` and derived `ConditionScript` and `EffectScript`, currently these just store a `NodeCPtr`.
- Added `ConditionalWeight`, implemented apart from support for `days` and `months` keys (which don't appear in vanilla), these will be added in another PR that adds support for keys with shared count limits.
- Added `Parser` cache to `Dataloader` and `parse_defines_cached`, used only for things which need to be stored for condition and effect parsing.
- Added script objects for every single script in the game, as well as code for storing each scripts' `NodeCPtr` in their object and triggering `parse_script` for every script object post-load.
- Added `allow_duplicates` argument to `NodeTools::expect_key`.
- Added some missing simple variables + code to load them from the defines.
- Minor style/format fixes (adding spaces after `if`, removing unused headers, etc.)

### CODE REVIEW

File: src/openvic-simulation/misc/Event.cpp
Comment (wvpm) [inline src/openvic-simulation/misc/Event.cpp:137]: ai_chance for event options, not ai_will_do

**Review by wvpm — DISMISSED** [review]
Approved with comments

**Comment by Hop311** [comment]
This won't merge until after #105 is merged and I've added the `parse_script` triggering code for all conditions and effects + fleshed out `ConditionalWeight`.

File: src/openvic-simulation/scripts/ConditionScript.hpp
Comment (Spartan322) [inline src/openvic-simulation/scripts/ConditionScript.hpp:8]:
```suggestion
	struct ConditionScript final : Script<GameManager const&> {
```

File: src/openvic-simulation/scripts/EffectScript.hpp
Comment (Spartan322) [inline src/openvic-simulation/scripts/EffectScript.hpp:8]:
```suggestion
	struct EffectScript final : Script<GameManager&> {
```

**Review by Spartan322 — COMMENTED** [review]
Always use final for virtual classes and methods you don't plan to extend further.

**Review by wvpm — APPROVED** [review]

**Pull request merged**
